platform/darwin/src/MGLPointCollection.mm

```diff
@@ -68,7 +68,7 @@ - (MGLCoordinateBounds)overlayBounds
 
 - (BOOL)intersectsOverlayBounds:(MGLCoordinateBounds)overlayBounds
 {
-    return MGLLatLngBoundsFromCoordinateBounds(_bounds).intersects(MGLLatLngBoundsFromCoordinateBounds(overlayBounds));
+    return MGLCoordinateBoundsIntersectsCoordinateBounds(_bounds, overlayBounds);
```

Comment:
This approach could also be used in `MGLMultiPolyline(gon) intersectsOverlayBounds:`. In general, I wonder if it'd ever be useful to add an `intersectsOverlay: <MGLOverlay>` API for all of these objects that conform to `MGLOverlay`?